frontend/src/metabase/lib/analytics.js

```diff
@@ -37,10 +44,39 @@ export const trackSchemaEvent = (schema, version, data) => {
   }
 
   if (Settings.snowplowEnabled()) {
-    trackSnowplowSchemaEvent(schema, version, data);
+    trackSnowplowSchemaEvent(schema, version, data, contextEntities);
   }
 };
 
+/**
+ * @param {string} name - Name of the action. Noun (target) + Verb in the past (action) which define the action taken - e.g. question-created, dashboard-updated, dashboard-auto-apply-filter-enabled
+ * @param {string} triggeredFrom - From where the action was taken. This can be generic like 'dashboard' or also more specific like 'dashboard_top_nav'.
+ * @param {string} targetId - (Optional) ID of the entity that the action was performed on. E.g. the ID of the question that was created in a question_created event.
+ * @param {number} durationMs - (Optional) Duration in milliseconds
+ * @param {string} result - (Optional) The outcome of the action (e.g. success, failure, …)
+ * @param {string} eventDetail - (Optional) Can be used for additional details that describe the event, e.g. the type of question that was created in a question_created event.
+ * @param {object} contextEntities - (Optional) Additional context entities (https://docs.snowplow.io/docs/understanding-your-pipeline/entities/). We send the Metabase Instance context by default with each event.
+ */
+export const trackActionEvent = (
+  name,
+  triggeredFrom,
+  targetId,
```

Comment:
I would put everything but the name in an object so that they become named parameters and we don't have to pass `null,null,null,xxx` when some of the optional fields are null.
Before going to prod with this I would also convert this to TS, but that's something a FE dev can take over

Reply:
1000% to everything @npretto said.

Reply:
Good suggestion. Should we convert the whole file to TS then or only move this `trackActionEvent` into a separate file?

Reply:
I am seeing complaints about the type checking now which I think we need to fix.
Would it make sense to actually turn the `trackSchemaEvent` arguments into an object as well?